modulus/metrics/diffusion/loss.py

```diff
@@ -546,6 +560,14 @@ def __call__(self, net, img_clean, img_lr, labels=None, augment_pipe=None):
                 self.patch_shape_y,
                 device=img_clean.device,
             )
+            global_index = torch.zeros(
+                b * self.patch_num,
+                2,
+                self.patch_shape_x,
+                self.patch_shape_y,
+                dtype=torch.int,
+                device=img_clean.device,
+            )
             for i in range(self.patch_num):
```

Comment:
**not for this PR since this code already exists**

Eventually, the loss should not know anything about patches and just treat them like batches (hey this rhymes). This can be achieved by moving the patching  logic either to the dataloader or training_loop.py. global_index can then be passed to the loss object.  Let's open an issue for this refactor.

Reply:
Agreed. However, it could be tricky since the patching operation should be done after the regression inference, while the regression inference is inside the loss function